src/racer/cargo.rs

```diff
@@ -705,8 +705,8 @@ pub fn get_crate_file(kratename: &str, from_path: &Path) -> Option<PathBuf> {
 
         let mut lockfile = tomlfile.clone();
         lockfile.pop();
-        if workspace.is_some() {
-            lockfile.push(workspace.unwrap());
+        if let Some(workspace) = workspace {
+            lockfile = workspace;
```

Comment:
Given that this behavior was surprising to both you and @jwilm, could you please add a comment here explaining that (including the link you provided in the PR?)

Reply:
Is that necessary considering we won't be using `push` like that anymore?